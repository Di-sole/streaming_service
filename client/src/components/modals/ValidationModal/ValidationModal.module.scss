.container {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;

    & h2 {
        color: rgba(255, 255, 255, 0.72);
        margin-bottom: 10px;
    }
}
