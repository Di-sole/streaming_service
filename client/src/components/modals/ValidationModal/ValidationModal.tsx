import ModalUI from '@/components/UI/Modal/ModalUI';
import styles from './ValidationModal.module.scss';
import { useRouter } from 'next/router';
import { useState } from 'react';
import TextLinkUI from '@/components/UI/links/TextLink/TextLinkUI';

const ValidationModal = () => {
    const { asPath, locale } = useRouter();
    const [isClose, setIsClose] = useState(false);
    const backPath = asPath.replace(/(\?validation)=/, '');

    return (
        <ModalUI className={styles.modal} close={isClose}>
            <div className={styles.container} data-testid={'validationModal'}>
                <h2>
                    {locale === 'ru'
                        ? 'Время сеанса истекло, пожалуйста, войдите снова!'
                        : 'Session expired, please login again!'}
                </h2>
                <TextLinkUI
                    href={backPath.includes('admin') ? `/?sign-in` : `${backPath}?sign-in`}
                    option="gradient"
                    onClick={() => setIsClose(true)}
                >
                    {locale === 'ru' ? 'Войти' : 'Log-in'}
                </TextLinkUI>
            </div>
        </ModalUI>
    );
};

export default ValidationModal;
