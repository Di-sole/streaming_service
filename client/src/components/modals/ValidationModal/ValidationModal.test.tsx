import { render, screen } from '@testing-library/react';
import ValidationModal from './ValidationModal';

jest.mock('next/router', () => ({
    useRouter() {
        return {
            route: '/',
            asPath: '/',
        };
    },
}));

describe('VALIDATIONMODAL SNAPSHOTS TESTS', () => {
    test('ValidationModal should not be changed', () => {
        render(<ValidationModal />);

        const validation = screen.getByTestId('validationModal');
        expect(validation).toMatchSnapshot();
    });
});
