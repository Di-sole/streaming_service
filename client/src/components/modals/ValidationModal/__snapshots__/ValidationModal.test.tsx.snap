// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VALIDATIONMODAL SNAPSHOTS TESTS ValidationModal should not be changed 1`] = `
<div
  class="container"
  data-testid="validationModal"
>
  <h2>
    Session expired, please login again!
  </h2>
  <a
    class="container container_gradient"
    href="/?sign-in"
  >
    Log-in
  </a>
</div>
`;
